fe/fe-core/src/main/java/org/apache/doris/task/AgentTaskQueue.java

```diff
@@ -150,6 +152,15 @@ public static synchronized List<AgentTask> getTask(long dbId, TTaskType type) {
         return signatureMap == null ? new ArrayList<>() : new ArrayList<>(signatureMap.values());
     }
 
+    public static synchronized void updateControlPublishVersion(long backendId,
+                                                                Map<TTaskType, Set<Long>> runningTasks) {
+        if (!tasks.containsRow(backendId) || !runningTasks.containsKey(TTaskType.PUBLISH_VERSION)) {
+            return;
+        }
+        Env.getCurrentSystemInfo().updateLastPublishVersionFailedMap(backendId,
+                runningTasks.get(TTaskType.PUBLISH_VERSION).size() > Config.publish_version_queued_limit_number);
```

Comment:
delete   tasks.containsRow(backendId)
because  when a txn finish,  fe will remove all be's publish task in agent queue. so task in fe may empty